src/index.js

```diff
@@ -0,0 +1,42 @@
+const fsp = require('fs').promises;
+const { readDataFiles } = require('./utils/data-parser');
+const { MatterParser } = require('./utils/matter-parser');
+
+async function getFileTimes(fileName, addCreated, addModified) {
+  const result = {};
+  if (!addCreated && !addModified) {
+    return result;
+  }
+  const stats = await fsp.stat(fileName);
+  if (addCreated) {
+    result.created = stats.birthtime.toISOString();
+  }
+  if (addModified) {
+    result.modified = stats.mtime.toISOString();
+  }
+  return result;
+}
+
+async function processFrontmatterFiles({
+  inputFilePatterns,
+  dataFilePatterns,
+  data = {},
+  addCreated = false,
+  addModified = false,
+}) {
+  const dataContents = await readDataFiles(dataFilePatterns);
+  const fileData = Object.assign.apply(this, [{}, ...dataContents]);
```

Comment:
Any reason you didn’t author it like this:
```suggestion
  const fileData = { ...dataContents }
```
Unless I’m missing what you are doing here?

Reply:
the result is an array of yaml data's from the input data files. For example: 'data.yaml' has `{"a":1}` inside, 'data.json' has `{"b":2}` inside.

The async `readDataFiles` function would return `[{"a": 1}, {"b": 2}]`. 
As it is an array, the `{ ...dataContents }` wouldn't merge the array of objects together into a new object, but creates an object of number properties:

`{ '0': { a: 1 }, '1': { b: 2 }}`